### DISCUSSION TITLE
Access and Change Alpine component data from external script tag

### DISCUSSION THREAD

**Discussion by Favourz1** [alpinejs/alpine#3632] [1. Help] [answered]
Hi, I'm using AlpineJs with AstroJs and i want to be able to manipulate and change the values of an alpinejs component from my script tag and also read the values.

I added alpineJs as a dependency to my project.

I tried using Alpine.data() but was unable to get through. Still getting value is undefined.

Example of what I'm trying to do

```
<div class="user-info" x-data="topbarState"> //throws an error topbarState is undefined 
              <div class="user-details">
                  <span class="user-name mb-1" x-text="fullName"></span> //it throws an error fullName and email  is undefined 
                  <span class="user-email" x-text="email"></span>
                  <span  @click="toggle">change mode</span>
            </div>
  </div>
```
```
<script>
    import Alpine from "alpinejs";

    const AUTHENTICATED_USER = JSON.parse(sessionStorage.getItem("usercredentials"));
    let isSuperAdminValue = "" // I want to be able to read the  value if isSuperAdmin in the AlpineJs component from here


    Alpine.data("topbarState", () => ({
         fullName: AUTHENTICATED_USER.fullName,
         email: AUTHENTICATED_USER.email,
         isSuperAdmin: false,
        toggle(){
            isSuperAdmin : !isSuperAdmin
        }
    }));
</script>`
```

Thank you in advance for your answers.
@calebporzio

**Comment by ekwoka** [comment]
oh, I think what you're looking for is to add, to the component just

```js
init() {
  Alpine.effect(() => isSuperAdminValue = this.isSuperAdmin)
}
```

**Reply by Favourz1** [reply]
No, this doesn't work

**Reply by ekwoka** [reply]
Never just say it doesn't work.

Show what you did, and explain what happened that you believe to be "not working".

What I showed DOES work, it may just not do what you want, because I don't know what you want, only what you explained.

**Reply by Favourz1** [reply, marked as answer]
Hey @ekwoka, Apologies didn't mean it that way.

I meant it didn't solve what I particularly wanted. Although your code is very correct.


I was missing `Alpine.start()`. Hence getting undefined as a value for `topbarState`. It works now.

Thanks again @ekwoka. I appreciate your answers and do not take it for granted!
